File: ejercicios/02_Squares_of_a_Sorted_Array/python/imista.py
def solution(nums):
    output = []
    mid = 0
    while(mid < len(nums)):
        if nums[mid] < 0:
            mid += 1
        else:
            break
    
    l = mid - 1
    r = mid

    while(l >= 0 and r < len(nums)):
        num_l = nums[l]
        num_r = nums[r]
        
        if(abs(num_l) > num_r):
            output.append(num_r**2)
            r += 1
        else:
            output.append(num_l**2)
            l -= 1

    while(l >= 0):
            output.append(nums[l]**2)
            l -= 1

    while(r < len(nums)):
        output.append(nums[r]**2)
        r += 1

    return output

File: ejercicios/02_Squares_of_a_Sorted_Array/javascript/imista.js
const solution = (nums) => {
    const SIZE = nums.length;
    const output = new Array(SIZE);
    let left = 0;
    let right = SIZE - 1;
    for (let i = SIZE - 1; i >= 0; i--) {
        if (Math.abs(nums[left]) < Math.abs(nums[right]))
            output[i] = nums[right--] ** 2;
        else
            output[i] = nums[left++] ** 2;
    }
    return output;
}

module.exports = { solution }